### DISCUSSION TITLE
HELP: Either it doesnt work or i am too stupid to use it

### DISCUSSION THREAD

**Discussion by BananenRitter** [SebastianMC/obsidian-custom-sort#120] [Q&A]
# What i want:
1. Place all Folders in the Rootfolder above all Files
- Sort all Files & Folders Alphanumerically (First Special Characters, then Numbers, then Letters)
2. Place all Files above all Folders in all Subfolders of Root
- Sort all Files Alphanumerically (First Special Characters, then Numbers, then Letters)
- Sort all Folders Alphanumerically

# My Syntax (Sortspec file location is in the Root Folder):
---
sorting-spec: |
    target-folder: /
    /+/:files/*
    < a-z
    /+/folders/*
    < a-z
    excalibrain ...
    sortspec ...
    ---

# It would be good to have a glossary witch all the symbols and their usecase in one location. The one already existing seems a bit scattered and incomplete

- Minimal Theme
# SYSTEM INFO:
	Obsidian version: v1.5.3
	Installer version: v1.4.16
	Operating system: Windows 10 Home 10.0.19045
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Base theme: dark
	Community theme: Minimal v7.4.6
	Snippets enabled: 9
	Restricted mode: off
	Plugins installed: 103
	Plugins enabled: 58
		1: Excalidraw v2.0.10
		2: Advanced Tables v0.19.1
		3: cMenu v1.1.2
		4: AutoMOC v1.3.4
		5: Buttons v0.4.19
		6: Google Calendar v1.10.14
		7: Calendar v1.5.10
		8: Surfing v0.8.16
		9: Citations v0.4.5
		10: Copy button for code blocks v0.1.0
		11: Dataview v0.5.64
		12: Dictionary v2.22.0
		13: Editor Syntax Highlight v0.1.3
		14: Excel to Markdown Table v0.4.0
		15: Find orphaned files and broken links v1.9.1
		16: Footnote Shortcut v0.1.3
		17: Kanban v1.5.3
		18: Markdown Formatting Assistant v0.4.1
		19: Pandoc Plugin v0.4.1
		20: Recent Files v1.3.8
		21: Tag Wrangler v0.6.1
		22: Zotero Integration v3.0.12
		23: Diagrams.net v1.0.4
		24: Diagrams v1.5.0
		25: Templater v2.0.0
		26: Image Toolkit v1.4.1
		27: Mousewheel Image zoom v1.0.19
		28: Periodic Notes v0.0.17
		29: Heatmap Calendar v0.6.0
		30: Icon Shortcodes v0.9.7
		31: Remember cursor position v1.0.8
		32: Show Current File Path v0.5.2
		33: MathLive v1.0.1
		34: Extended MathJax v0.4.1
		35: Go To Heading v0.2.2
		36: Export To TeX v0.4.2
		37: Quick Latex for Obsidian v2.6.3
		38: Completr v3.2.0
		39: Latex Suite v1.8.10
		40: ExcaliBrain v0.2.11
		41: Edit History v0.1.3
		42: Automatic Table Of Contents v1.0.6
		43: Navigate Cursor History v1.1.0
		44: Keyboard Analyzer v0.1.3
		45: Hotkey Helper v0.3.18
		46: Metadata Menu v0.7.0
		47: MetaEdit v1.8.2
		48: Projects v1.17.1
		49: Minimal Theme Settings v7.3.1
		50: Style Settings v1.0.7
		51: Callout Integrator v1.1.4
		52: List Callouts v1.2.0
		53: Callout Manager v1.0.1
		54: Trash Explorer v1.2.2
		55: Media Extended v2.11.1
		56: Multi-Column Markdown v0.8.3
		57: HTML Tabs v1.1.0
		58: Custom File Explorer sorting v2.1.2

**Comment by SebastianMC** [comment]
Hi @BananenRitter 

I agree with your remark on the documentation being

> ... a bit scattered and incomplete

As the plugin evolved and new functions were added, the syntax was getting richer and richer (while keeping the backward compatibility), I struggled with maintaining the documentation within a single extremely large `README.md`. I was seeking for more convenient formats for non-trivial multi-pages multi-sections (plus videos) documentation. Easy to consume, navigate and search, and also easy to create and maintain. Eventually I decided (just recently) to take the same approach as the famous Obsidian dataview plugin. The documentation site is being created & migrated at https://www.obsidian-custom-sort.info (empty at now).

Regarding your specific request, please try the below sorting specification:

```yaml
---
sorting-spec: |
  target-folder: /
  /folders
    < a-z
  /:files
    < a-z
  excalibrain...
  sortspec
  
  target-folder: /*
  /:files
    < a-z
  /folders
    < a-z
---
```

The meaning of `  excalibrain ...` entry was unclear for me, so I transformed it to catch all files and folders which name starts with (or is equal to)`excalibrain`

The empty line in the middle of the specification is optional. I inserted it for readability to visually separate sorting spec for the root folder `  target-folder: /` from specification of sorting for all descendant folders `  target-folder: /*`. A technical note: if the explicit specification for the root folder (`  target-folder: /`) would not exist, the specification for descendant folders (`  target-folder: /*`) would also be applied to the root folder.

Let me know it if works on your side. I'm also happy to explain some details of this specific sorting configuration, if you have questions.
